lib/cpp/CMakeLists.txt

```diff
@@ -90,9 +89,9 @@ else()
     )
 endif()
 
-# If OpenSSL is not found just ignore the OpenSSL stuff
-find_package(OpenSSL)
-if(OPENSSL_FOUND AND WITH_OPENSSL)
+if(WITH_OPENSSL)
+    hunter_add_package(OpenSSL)
+    find_package(OpenSSL REQUIRED)
```

Comment:
Imported targets `OpenSSL::SSL`/`OpenSSL::Crypto` should be used here, otherwise package will not be relocatable.